wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -62,6 +62,17 @@ public static synchronized CommandScheduler getInstance() {
     return instance;
   }
 
+  /**
+   * Resets the Scheduler instance, which is useful for testing purposes. This should not be called
+   * from user code.
+   */
+  public static synchronized void resetInstance() {
```

Comment:
Sounds like a good idea- How could we detect we're running on a real robot versus in tests? I'd assume we wouldn't want to use `isFMSAttached()` because then it might lead to code working in the lab but not in an actual match, which would really suck.

I guess the way to really *really* avoid sneaky behavior changes is to always check (somehow) whether the call is from robot code (and not test code), but that's going to be really tricky for C++.

Reply:
`isReal()`/`isSimulation()`?